### PR TITLE
docs: convert code block to typescript

### PR DESCRIPTION
**Pull request by wsuwt**
## Description

- add TypeScript to code blocks
- display code blocks in both JavaScript and TypeScript.

### CODE REVIEW

File: documents/src/pages/elements/button-bar.md [hunk 1]
```diff
@@ -216,10 +218,22 @@ To listen to the tap event on the button, add the `tap` event listener to an ind
   <ef-button id="Like" icon="like-empty"></ef-button>
 </ef-button-bar>
 ```
+
 ```javascript
 const buttonBar = document.getElementById('button-bar');
```
Comment (Theeraphat-Sorasetsakul) [inline documents/src/pages/elements/button-bar.md]:
```suggestion
const buttonBar = document.querySelector('ef-button-bar');
```

File: documents/src/pages/elements/radio-button.md [hunk 2]
```diff
@@ -70,12 +71,23 @@ More than one `ef-radio-button` can be grouped by setting the same value to the
...
 ## Handle check value changed
+
 `checked-changed` is the **only** event fired by `ef-radio-button`. It is dispatched whenever the state has been changed by user interaction, such as a click, tap or keyboard event.
 
 ```javascript
-radioButtonGroup.addEventListener('checked-changed', (e) => {
-  if (e.target.checked) {
-    // console.log(e.target.id);
+radioButtonGroup.addEventListener('checked-changed', (event) => {
+  if (event.target.checked) {
```
Comment (Theeraphat-Sorasetsakul) [inline documents/src/pages/elements/radio-button.md:79]: If condition content  doesn't same as TS.

File: documents/src/pages/elements/combo-box.md [hunk 3]
```diff
@@ -258,7 +319,7 @@ Finally, listen for the `query-changed` event to make calls to the server and se
 
 ```javascript
 combo.addEventListener('query-changed', async () => {
```
Comment (Theeraphat-Sorasetsakul) [inline documents/src/pages/elements/combo-box.md]:
```suggestion
comboBox.addEventListener('query-changed', async () => {
```

File: documents/src/pages/elements/combo-box.md [hunk 4]
```diff
@@ -362,32 +423,73 @@ const flagMap = new WeakMap();
...
+import '@refinitiv-ui/elements/flag';
+import '@refinitiv-ui/elements/flag/themes/halo/dark';
+
+// Keep the reference to the default renderer
+const itemRenderer = new ComboBoxRenderer(comboBox);
+// Keep track flag elements after creating to avoid memory leak and re-render the same flag
+const flagMap = new WeakMap();
 
+// Create a re-useable renderer that shows Flags next to the country
+comboBox.renderer = (item: ItemData, composer: CollectionComposer, element: ListItem) => {
```
Comment (Theeraphat-Sorasetsakul) [inline documents/src/pages/elements/combo-box.md:469]: Missing import `ItemData`. Pls add it.

File: documents/src/pages/elements/radio-button.md (hunk 2 above)
Comment (wsuwt) [reply documents/src/pages/elements/radio-button.md:79]: Updated.

File: documents/src/pages/elements/combo-box.md (hunk 4 above)
Comment (wsuwt) [reply documents/src/pages/elements/combo-box.md:469]: Updated.

**Review by Theeraphat-Sorasetsakul — APPROVED** [review]
LGTM.

**Pull request merged**